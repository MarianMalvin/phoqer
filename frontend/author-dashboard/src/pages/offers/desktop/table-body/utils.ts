import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'category', title: 'Category' },
    { key: 'reviews', title: 'Reviews' },
    { key: 'price', title: 'Price' },
    { key: 'sale_percentage', title: 'Sale percentage' },
    { key: 'sale_description', title: 'Sale description' },
    { key: 'description', title: 'Description' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    images: TableCellSizeEnum.Small2Extra,
    title: TableCellSizeEnum.Large5Extra,
    description: TableCellSizeEnum.Large8Extra,
    sale_percentage: TableCellSizeEnum.MiddleExtra,
    sale_description: TableCellSizeEnum.Large4Extra,
    reviews: TableCellSizeEnum.Middle,
    price: TableCellSizeEnum.Middle,
    category: TableCellSizeEnum.Middle,
};
