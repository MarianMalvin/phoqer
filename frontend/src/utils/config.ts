import { Themes } from '../interfaces';

const host = process.env.NODE_ENV === 'development' ? 'http://dev.phoqer.com' : '';
const api = host + '/api/';

const config = {
    baseUrl: {
        development: (version = 'v1'): string => api + version,
        test: (version = 'v1'): string => api + version,
        production: (version = 'v1'): string => (process.browser ? api + version : 'http://backend:8001'),
    },
    socketUrl: (version = 'v1'): string => 'ws://' + (api.split('://')[1] || 'phoqer.com/api/') + version,
    uploadsUrl: (version = 'v2'): string => `${api}${version}/upload`,
    host: (lang = 'pl'): string => `${host}/${lang}`,
    img: host,
    themes: ['white', 'black'] as Themes[],
    googleApiKey: process.env.NEXT_PUBLIC_GOOGLE_MAP_API_KEY,
    appSecretKey: process.env.NEXT_PUBLIC_SECRET_KEY,
};

export default config;
