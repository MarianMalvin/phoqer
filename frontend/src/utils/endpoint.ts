const endpointsMap = {
    ssr: (path = ''): string => (process.env.NEXT_PUBLIC_API_SSR || 'http://dev.phoqer.com/api/v2') + path,
    browser: (path = ''): string => (process.env.NEXT_PUBLIC_API_BROWSER || '') + path,
};

const endpoint = endpointsMap[process.browser ? 'browser' : 'ssr'];

export default endpoint;
