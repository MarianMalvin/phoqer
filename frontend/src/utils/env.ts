const env = {
    NEXT_PUBLIC_URL: process.env.NEXT_PUBLIC_URL || 'http://dev.phoqer.com',
    NEXT_PUBLIC_WS: process.env.NEXT_PUBLIC_WS || 'ws://dev.phoqer.com/api/v1',
    NEXT_PUBLIC_GOOGLE_MAP_API_KEY: process.env.NEXT_PUBLIC_GOOGLE_MAP_API_KEY || '',
    NEXT_PUBLIC_SECRET_KEY: process.env.NEXT_PUBLIC_SECRET_KEY || '',
};

export default env;
