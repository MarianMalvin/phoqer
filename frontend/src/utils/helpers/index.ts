export { mailRegex, passwordRegex, numberValidation, isEmpty } from './validation';
export { addZeroToNumber, moneyFormat, declOfNum } from './formatting';
export { dateFromTimestamp, addMonthToDate, formatTimestamp } from './date';
export { cutString } from './string';
export { throttle } from './lodash';
export { parseCookie, serverCookie, serverRedirect } from './ssr';
export { logger } from './logger';
export { formatCatList, findCategory, findSubCategory, findParentCategory } from './categories';
export { onlineStatus, timeIndicator } from './online';
