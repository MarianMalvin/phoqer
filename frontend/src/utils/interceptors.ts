import axios from 'axios';
import * as AxiosLogger from 'axios-logger';

const api = axios.create({});

api.interceptors.request.use(
    req => AxiosLogger.requestLogger(req, { data: false }),
    error => Promise.reject(error),
);

api.interceptors.response.use(
    res => AxiosLogger.responseLogger(res, { data: false }),
    error => Promise.reject(error),
);

export default api;
