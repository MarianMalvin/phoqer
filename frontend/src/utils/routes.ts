import { ChatTypeEnum } from '../interfaces';

type ID = string | number | null;
type Primitive = string | number;

const routes = {
    root: '/',
    auth: {
        login: '/auth/login',
        join: '/auth/join',
        forgot_pass: '/auth/forgot-pass',
        confirmation: '/auth/confirmation',
    },
    profile: {
        public: (id: ID = ':profileId'): string => `/profile/${id}`,
        private: '/profile/private',
    },
    my_offers: (status = 'all'): string => `/my-offers/${status}`,

    chat: {
        list: (type: ChatTypeEnum = ChatTypeEnum.CLIENT): string => `/chat/${type}`,
        item: (chat: Primitive, type: ChatTypeEnum = ChatTypeEnum.CLIENT): string => `/chat/${type}/${chat}`,
        new: (offerId: Primitive): string => `/chat/new/${offerId}`,
    },
    settings: (status: Primitive = 'general'): string => `/settings/${status}`,
    notifications: '/notifications',
    referral: '/referral',
    analytics: '/analytics',
    offers: {
        single: (id: ID = ':offerId'): string => `/offers/${id}`,
        edit: (id: ID = ':offerId'): string => `/offers/edit/${id}`,
        new: (step: ID = ':step', offerId?: string): string => `/offers/new/${step}${offerId ? '?offerId=' + offerId : ''}`,
        list: '/offers',
    },
    static: {
        help: '/static/help',
        business: '/static/business',
        rules: '/static/rules',
        politic: '/static/politic',
        faq: '/static/faq',
        safety: '/static/safety',
        map: '/static/map',
    },
    favorite: '/favorite',
};

export default routes;
