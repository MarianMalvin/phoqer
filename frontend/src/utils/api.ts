import axios, { AxiosResponse } from 'axios';

import {
    ChatType,
    IAuth,
    ICategories,
    IChats,
    IComment,
    IMessages,
    IOfferCard,
    IOfferDynamic,
    IOfferPagination,
    IPagination,
    IPublicProfile,
    ISignup,
} from '../interfaces';
import { INewChat } from '../redux/chat/chats/interfaces';
import { IBody } from '../redux/offers/new_offer/interfaces';

import config from './config';

const url2 = config.baseUrl[process.env.NODE_ENV]('v2');

interface ICommentBody {
    body: string;
    replies_id?: number;
    offer_id?: string;
    images: string[];
}

const api = {
    // V2
    categories: {
        get: (): Promise<AxiosResponse<ICategories>> => axios.get(`${url2}/categories`),
    },
    auth: {
        login: (form: FormData): Promise<AxiosResponse<IAuth>> => axios.post(`${url2}/auth/login`, form),
        signup: (body: ISignup): Promise<AxiosResponse<void>> => axios.post(`${url2}/users/signup`, body),
    },
    comments: {
        list: (id: string): Promise<AxiosResponse<IComment[]>> => axios.get(`${url2}/comments/${id}`),
        create: (body: ICommentBody): Promise<AxiosResponse<void>> => axios.post(`${url2}/comments`, body),
        delete: (id: number): Promise<AxiosResponse<void>> => axios.delete(`${url2}/comments/${id}`),
        like: (id: number): Promise<AxiosResponse<void>> => axios.patch(`${url2}/comments/${id}/like`),
        dislike: (id: number): Promise<AxiosResponse<void>> => axios.patch(`${url2}/comments/${id}/dislike`),
    },
    offers: {
        myOffers: ({ tab, params }: { tab: string; params: { [key: string]: unknown } }): Promise<AxiosResponse<IOfferDynamic>> =>
            axios.get(`${url2}/offers/status/${tab}`, { params }),
        publicOffers: (id: string, params: { [key: string]: unknown }): Promise<AxiosResponse<IOfferPagination>> =>
            axios.get(`${url2}/offers/public/${id}`, { params }),
        popular: (): Promise<AxiosResponse<IOfferPagination>> => axios.get(`${url2}/offers/popular`),
        single: (id: string): Promise<AxiosResponse<IOfferCard>> => axios.get(`${url2}/offers/${id}`),
        new: (body: IBody): Promise<AxiosResponse<void>> => axios.post(`${url2}/offers`, body),
        update: (id: string, body: IBody | Partial<Body>): Promise<AxiosResponse<void>> =>
            axios.patch(`${url2}/offers/${id}`, body),
        status: (id: string, body: { [key: string]: unknown }): Promise<AxiosResponse<void>> =>
            axios.patch(`${url2}/offers/status/${id}`, body),
        search: (params: { [key: string]: unknown }): Promise<AxiosResponse<IOfferPagination>> =>
            axios.get(`${url2}/offers/search`, { params }),
        deleteOffer: (id: string): Promise<AxiosResponse<void>> => axios.delete(`${url2}/offers/${id}`),
        favorite: {
            get: (): Promise<AxiosResponse<IOfferCard[]>> => axios.get(`${url2}/favorite`),
            patch: (id: string): Promise<AxiosResponse<IOfferCard[]>> => axios.patch(`${url2}/favorite/${id}`),
        },
    },
    profiles: {
        private: {
            user: (): Promise<AxiosResponse<IPublicProfile>> => axios.get(`${url2}/users/me`),
            userUpdate: (body: Partial<IPublicProfile>): Promise<AxiosResponse<IPublicProfile>> =>
                axios.patch(`${url2}/users/me`, body),
        },
        public: {
            get: (id: number): Promise<AxiosResponse<IPublicProfile>> => axios.get(`${url2}/users/${id}`),
            userShort: (id: number): Promise<AxiosResponse<IPublicProfile>> => axios.get(`${url2}/users/short/${id}`),
        },
    },
    chat: {
        chats: (page = 1, type: ChatType = 'i_am_client'): Promise<AxiosResponse<IPagination<IChats>>> =>
            axios.get(`${url2}/chats?${type}=true`, { params: { page } }),
        createChat: (body: INewChat): Promise<AxiosResponse<INewChat>> => axios.post(`${url2}/chats`, body),
        messages: (id: number, page = 1): Promise<AxiosResponse<IPagination<IMessages>>> =>
            axios.get(`${url2}/chats/${id}`, { params: { page } }),
        offerInfo: (id: number): Promise<AxiosResponse<IOfferCard>> => axios.get(`${url2}/offers/offers/${id}`),
    },
    locations: {
        countries: (): Promise<AxiosResponse<void>> => axios.get(`${url2}/locations/countries`),
        cities: (slug: string): Promise<AxiosResponse<void>> => axios.get(`${url2}/locations/cities/${slug}`),
    },
    uploads: (form: FormData): Promise<AxiosResponse<{ image_url: string }>> =>
        axios.post(`${url2}/upload`, form, { headers: { 'content-type': 'multipart/form-data' } }),
};

export default api;
