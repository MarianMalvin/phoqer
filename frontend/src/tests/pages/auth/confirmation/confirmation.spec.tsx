import Confirmation from '../../../../pages/auth/confirmation';
import { render, waitFor } from '../../../tests.utils';

describe('Confirmation component', () => {
    it('match snapshot', async () => {
        const { asFragment } = render(<Confirmation />);
        const lazyContent = await waitFor(() => asFragment());
        expect(lazyContent).toMatchSnapshot();
    });
});
