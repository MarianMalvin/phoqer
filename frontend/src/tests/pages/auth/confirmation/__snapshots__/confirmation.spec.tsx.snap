// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation component match snapshot 1`] = `
<DocumentFragment>
  <h1
    class="title-0-2-8"
  >
    PHOQER is an online platform that brings people together to share goods and services. PHOQER - here you will find what you are looking for! By clicking the Submit button, you will be able to place offers on any topic easily and quickly.
  </h1>
  <div
    class="backdrop-0-2-15"
  />
  <header
    class="header-0-2-9"
    tabindex="0"
  >
    <div
      class="container-0-2-16"
    >
      <div
        class="flex-0-2-11"
        tabindex="0"
      >
        <div
          class="wrp-0-2-12"
        >
          <a
            class="logo-0-2-17"
            href="/"
          >
            <svg
              class="img-0-2-20"
              fill="none"
              viewBox="0 0 148 39"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.3676 8.6C14.3516 8.6 16.069 8.93067 17.5196 9.592C18.9916 10.2533 20.1223 11.192 20.9116 12.408C21.701 13.624 22.0956 15.064 22.0956 16.728C22.0956 18.3707 21.701 19.8107 20.9116 21.048C20.1223 22.264 18.9916 23.2027 17.5196 23.864C16.069 24.504 14.3516 24.824 12.3676 24.824H7.85563V31H2.67163V8.6H12.3676ZM12.0796 20.6C13.637 20.6 14.821 20.2693 15.6316 19.608C16.4423 18.9253 16.8476 17.9653 16.8476 16.728C16.8476 15.4693 16.4423 14.5093 15.6316 13.848C14.821 13.1653 13.637 12.824 12.0796 12.824H7.85563V20.6H12.0796ZM46.3406 8.6V31H41.1566V21.816H30.9806V31H25.7966V8.6H30.9806V17.432H41.1566V8.6H46.3406ZM62.4884 31.384C60.163 31.384 58.0617 30.8827 56.1844 29.88C54.3284 28.8773 52.867 27.5013 51.8004 25.752C50.755 23.9813 50.2324 21.9973 50.2324 19.8C50.2324 17.6027 50.755 15.6293 51.8004 13.88C52.867 12.1093 54.3284 10.7227 56.1844 9.72C58.0617 8.71733 60.163 8.216 62.4884 8.216C64.8137 8.216 66.9044 8.71733 68.7604 9.72C70.6164 10.7227 72.0777 12.1093 73.1444 13.88C74.211 15.6293 74.7444 17.6027 74.7444 19.8C74.7444 21.9973 74.211 23.9813 73.1444 25.752C72.0777 27.5013 70.6164 28.8773 68.7604 29.88C66.9044 30.8827 64.8137 31.384 62.4884 31.384ZM62.4884 26.968C63.811 26.968 65.0057 26.6693 66.0724 26.072C67.139 25.4533 67.971 24.6 68.5684 23.512C69.187 22.424 69.4964 21.1867 69.4964 19.8C69.4964 18.4133 69.187 17.176 68.5684 16.088C67.971 15 67.139 14.1573 66.0724 13.56C65.0057 12.9413 63.811 12.632 62.4884 12.632C61.1657 12.632 59.971 12.9413 58.9044 13.56C57.8377 14.1573 56.995 15 56.3764 16.088C55.779 17.176 55.4804 18.4133 55.4804 19.8C55.4804 21.1867 55.779 22.424 56.3764 23.512C56.995 24.6 57.8377 25.4533 58.9044 26.072C59.971 26.6693 61.1657 26.968 62.4884 26.968ZM102.864 33.4C102.139 34.296 101.254 34.9787 100.208 35.448C99.1844 35.9173 98.0537 36.152 96.8164 36.152C95.1524 36.152 93.6484 35.7893 92.3044 35.064C90.9604 34.36 89.4244 33.1013 87.6964 31.288C85.6697 31.032 83.8564 30.3813 82.2564 29.336C80.6777 28.2907 79.4404 26.9467 78.5444 25.304C77.6697 23.64 77.2324 21.8053 77.2324 19.8C77.2324 17.6027 77.755 15.6293 78.8004 13.88C79.867 12.1093 81.3284 10.7227 83.1844 9.72C85.0617 8.71733 87.163 8.216 89.4884 8.216C91.8137 8.216 93.9044 8.71733 95.7604 9.72C97.6164 10.7227 99.0777 12.1093 100.144 13.88C101.211 15.6293 101.744 17.6027 101.744 19.8C101.744 22.4027 101.008 24.6853 99.5364 26.648C98.0857 28.6107 96.155 29.976 93.7444 30.744C94.2777 31.2987 94.7897 31.6933 95.2804 31.928C95.7924 32.184 96.3364 32.312 96.9124 32.312C98.299 32.312 99.515 31.7573 100.56 30.648L102.864 33.4ZM82.4804 19.8C82.4804 21.1867 82.779 22.424 83.3764 23.512C83.995 24.6 84.8377 25.4533 85.9044 26.072C86.971 26.6693 88.1657 26.968 89.4884 26.968C90.811 26.968 92.0057 26.6693 93.0724 26.072C94.139 25.4533 94.971 24.6 95.5684 23.512C96.187 22.424 96.4964 21.1867 96.4964 19.8C96.4964 18.4133 96.187 17.176 95.5684 16.088C94.971 15 94.139 14.1573 93.0724 13.56C92.0057 12.9413 90.811 12.632 89.4884 12.632C88.1657 12.632 86.971 12.9413 85.9044 13.56C84.8377 14.1573 83.995 15 83.3764 16.088C82.779 17.176 82.4804 18.4133 82.4804 19.8ZM122.984 26.84V31H105.64V8.6H122.568V12.76H110.792V17.624H121.192V21.656H110.792V26.84H122.984ZM141.381 31L137.061 24.76H136.805H132.293V31H127.109V8.6H136.805C138.789 8.6 140.506 8.93067 141.957 9.592C143.429 10.2533 144.56 11.192 145.349 12.408C146.138 13.624 146.533 15.064 146.533 16.728C146.533 18.392 146.128 19.832 145.317 21.048C144.528 22.2427 143.397 23.16 141.925 23.8L146.949 31H141.381ZM141.285 16.728C141.285 15.4693 140.88 14.5093 140.069 13.848C139.258 13.1653 138.074 12.824 136.517 12.824H132.293V20.632H136.517C138.074 20.632 139.258 20.2907 140.069 19.608C140.88 18.9253 141.285 17.9653 141.285 16.728Z"
              />
            </svg>
            <small
              class="label-0-2-18"
            >
              alpha
            </small>
          </a>
        </div>
        <div
          class="wrp-0-2-12"
        >
          <div
            class="wrp-0-2-21"
          >
            <p
              class="text-0-2-22"
            >
              EN
            </p>
            <div
              class="wrp-0-2-24 lang-0-2-23"
            >
              <button
                class="inner-0-2-25"
                style="height: 3.5rem;"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-globe-americas fa-w-16 "
                  data-icon="globe-americas"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 496 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm82.29 357.6c-3.9 3.88-7.99 7.95-11.31 11.28-2.99 3-5.1 6.7-6.17 10.71-1.51 5.66-2.73 11.38-4.77 16.87l-17.39 46.85c-13.76 3-28 4.69-42.65 4.69v-27.38c1.69-12.62-7.64-36.26-22.63-51.25-6-6-9.37-14.14-9.37-22.63v-32.01c0-11.64-6.27-22.34-16.46-27.97-14.37-7.95-34.81-19.06-48.81-26.11-11.48-5.78-22.1-13.14-31.65-21.75l-.8-.72a114.792 114.792 0 0 1-18.06-20.74c-9.38-13.77-24.66-36.42-34.59-51.14 20.47-45.5 57.36-82.04 103.2-101.89l24.01 12.01C203.48 89.74 216 82.01 216 70.11v-11.3c7.99-1.29 16.12-2.11 24.39-2.42l28.3 28.3c6.25 6.25 6.25 16.38 0 22.63L264 112l-10.34 10.34c-3.12 3.12-3.12 8.19 0 11.31l4.69 4.69c3.12 3.12 3.12 8.19 0 11.31l-8 8a8.008 8.008 0 0 1-5.66 2.34h-8.99c-2.08 0-4.08.81-5.58 2.27l-9.92 9.65a8.008 8.008 0 0 0-1.58 9.31l15.59 31.19c2.66 5.32-1.21 11.58-7.15 11.58h-5.64c-1.93 0-3.79-.7-5.24-1.96l-9.28-8.06a16.017 16.017 0 0 0-15.55-3.1l-31.17 10.39a11.95 11.95 0 0 0-8.17 11.34c0 4.53 2.56 8.66 6.61 10.69l11.08 5.54c9.41 4.71 19.79 7.16 30.31 7.16s22.59 27.29 32 32h66.75c8.49 0 16.62 3.37 22.63 9.37l13.69 13.69a30.503 30.503 0 0 1 8.93 21.57 46.536 46.536 0 0 1-13.72 32.98zM417 274.25c-5.79-1.45-10.84-5-14.15-9.97l-17.98-26.97a23.97 23.97 0 0 1 0-26.62l19.59-29.38c2.32-3.47 5.5-6.29 9.24-8.15l12.98-6.49C440.2 193.59 448 223.87 448 256c0 8.67-.74 17.16-1.82 25.54L417 274.25z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="root-0-2-36"
          >
            <button
              class="btn-0-2-37"
              type="button"
            >
              Login
            </button>
            <button
              class="btn-0-2-37"
              type="button"
            >
              Join Phoqer
            </button>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    class="root-0-2-1"
  >
    <div
      class="container-0-2-16"
    >
      <div
        class="flex-0-2-2"
      >
        <div
          class="row-0-2-3"
        >
          <h2
            class="title-0-2-5"
          >
            Thank you for registration!
          </h2>
          <p
            class="text-0-2-6"
          >
            Лол кек чебурек Лол кек чебурек. Лол кек чебурек Лол кек чебурек Лол кек чебурекЛол кек чебурек
          </p>
          <div
            class="flex-0-2-2"
          >
            <a
              class="btn-0-2-4"
              href="/"
            >
              To home page
            </a>
            <button
              class="btn-0-2-4"
            >
              Login
            </button>
          </div>
        </div>
      </div>
      <div
        class="logo-0-2-7"
      >
        <svg
          fill="none"
          viewBox="0 0 148 39"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.3676 8.6C14.3516 8.6 16.069 8.93067 17.5196 9.592C18.9916 10.2533 20.1223 11.192 20.9116 12.408C21.701 13.624 22.0956 15.064 22.0956 16.728C22.0956 18.3707 21.701 19.8107 20.9116 21.048C20.1223 22.264 18.9916 23.2027 17.5196 23.864C16.069 24.504 14.3516 24.824 12.3676 24.824H7.85563V31H2.67163V8.6H12.3676ZM12.0796 20.6C13.637 20.6 14.821 20.2693 15.6316 19.608C16.4423 18.9253 16.8476 17.9653 16.8476 16.728C16.8476 15.4693 16.4423 14.5093 15.6316 13.848C14.821 13.1653 13.637 12.824 12.0796 12.824H7.85563V20.6H12.0796ZM46.3406 8.6V31H41.1566V21.816H30.9806V31H25.7966V8.6H30.9806V17.432H41.1566V8.6H46.3406ZM62.4884 31.384C60.163 31.384 58.0617 30.8827 56.1844 29.88C54.3284 28.8773 52.867 27.5013 51.8004 25.752C50.755 23.9813 50.2324 21.9973 50.2324 19.8C50.2324 17.6027 50.755 15.6293 51.8004 13.88C52.867 12.1093 54.3284 10.7227 56.1844 9.72C58.0617 8.71733 60.163 8.216 62.4884 8.216C64.8137 8.216 66.9044 8.71733 68.7604 9.72C70.6164 10.7227 72.0777 12.1093 73.1444 13.88C74.211 15.6293 74.7444 17.6027 74.7444 19.8C74.7444 21.9973 74.211 23.9813 73.1444 25.752C72.0777 27.5013 70.6164 28.8773 68.7604 29.88C66.9044 30.8827 64.8137 31.384 62.4884 31.384ZM62.4884 26.968C63.811 26.968 65.0057 26.6693 66.0724 26.072C67.139 25.4533 67.971 24.6 68.5684 23.512C69.187 22.424 69.4964 21.1867 69.4964 19.8C69.4964 18.4133 69.187 17.176 68.5684 16.088C67.971 15 67.139 14.1573 66.0724 13.56C65.0057 12.9413 63.811 12.632 62.4884 12.632C61.1657 12.632 59.971 12.9413 58.9044 13.56C57.8377 14.1573 56.995 15 56.3764 16.088C55.779 17.176 55.4804 18.4133 55.4804 19.8C55.4804 21.1867 55.779 22.424 56.3764 23.512C56.995 24.6 57.8377 25.4533 58.9044 26.072C59.971 26.6693 61.1657 26.968 62.4884 26.968ZM102.864 33.4C102.139 34.296 101.254 34.9787 100.208 35.448C99.1844 35.9173 98.0537 36.152 96.8164 36.152C95.1524 36.152 93.6484 35.7893 92.3044 35.064C90.9604 34.36 89.4244 33.1013 87.6964 31.288C85.6697 31.032 83.8564 30.3813 82.2564 29.336C80.6777 28.2907 79.4404 26.9467 78.5444 25.304C77.6697 23.64 77.2324 21.8053 77.2324 19.8C77.2324 17.6027 77.755 15.6293 78.8004 13.88C79.867 12.1093 81.3284 10.7227 83.1844 9.72C85.0617 8.71733 87.163 8.216 89.4884 8.216C91.8137 8.216 93.9044 8.71733 95.7604 9.72C97.6164 10.7227 99.0777 12.1093 100.144 13.88C101.211 15.6293 101.744 17.6027 101.744 19.8C101.744 22.4027 101.008 24.6853 99.5364 26.648C98.0857 28.6107 96.155 29.976 93.7444 30.744C94.2777 31.2987 94.7897 31.6933 95.2804 31.928C95.7924 32.184 96.3364 32.312 96.9124 32.312C98.299 32.312 99.515 31.7573 100.56 30.648L102.864 33.4ZM82.4804 19.8C82.4804 21.1867 82.779 22.424 83.3764 23.512C83.995 24.6 84.8377 25.4533 85.9044 26.072C86.971 26.6693 88.1657 26.968 89.4884 26.968C90.811 26.968 92.0057 26.6693 93.0724 26.072C94.139 25.4533 94.971 24.6 95.5684 23.512C96.187 22.424 96.4964 21.1867 96.4964 19.8C96.4964 18.4133 96.187 17.176 95.5684 16.088C94.971 15 94.139 14.1573 93.0724 13.56C92.0057 12.9413 90.811 12.632 89.4884 12.632C88.1657 12.632 86.971 12.9413 85.9044 13.56C84.8377 14.1573 83.995 15 83.3764 16.088C82.779 17.176 82.4804 18.4133 82.4804 19.8ZM122.984 26.84V31H105.64V8.6H122.568V12.76H110.792V17.624H121.192V21.656H110.792V26.84H122.984ZM141.381 31L137.061 24.76H136.805H132.293V31H127.109V8.6H136.805C138.789 8.6 140.506 8.93067 141.957 9.592C143.429 10.2533 144.56 11.192 145.349 12.408C146.138 13.624 146.533 15.064 146.533 16.728C146.533 18.392 146.128 19.832 145.317 21.048C144.528 22.2427 143.397 23.16 141.925 23.8L146.949 31H141.381ZM141.285 16.728C141.285 15.4693 140.88 14.5093 140.069 13.848C139.258 13.1653 138.074 12.824 136.517 12.824H132.293V20.632H136.517C138.074 20.632 139.258 20.2907 140.069 19.608C140.88 18.9253 141.285 17.9653 141.285 16.728Z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
